UnitySDK/Assets/ML-Agents/Scripts/RewardProvider/IRewardProvider.cs

```diff
@@ -0,0 +1,15 @@
+namespace MLAgents.RewardProvider
+{
+    /// <summary>
+    /// Reward providers allow users to provide rewards for Agent behavior during training in order to
+    /// give hints on what types of actions are "better" than others based on an Agent's previous observation.
+    /// </summary>
+    public interface IRewardProvider
```

Comment:
IRewardProvider probably also need a way to signal stepping : When the reward has been collected, the incremental reward must be reset. Unless GetIncrementalReward also resets the incremental reward?

Reply:
I had some code to step reward providers in the Agent.  I think I like that better than having GetIncrementalReward do two things.

Reply:
I'll add it back and ping you for a review on it

Reply:
@vincentpierre I've added it in the latest commit.  Still feels a bit clunky having ResetReward in Agent.  But I think it's better than the alternative of subclassing Agent in all of our environments to account for this.